### DISCUSSION TITLE
[@types/ws] Argument of type 'Duplex' is not assignable to parameter of type 'Socket'

### DISCUSSION THREAD

**Discussion by strmer15** [DefinitelyTyped/DefinitelyTyped#55095] [Issues with a @types package]
I just upgraded to the latest `@types/node` and now I'm getting this error:

```
src/index.ts:106:32 - error TS2345: Argument of type 'Duplex' is not assignable to parameter of type 'Socket'.
  Type 'Duplex' is missing the following properties from type 'Socket': connect, setTimeout, setNoDelay, setKeepAlive, and 9 more.

106     wss.handleUpgrade(request, socket, head, function (ws) {
                                   ~~~~~~
```

See https://github.com/websockets/ws/issues/1926 as well for another developer with the same issue who opened a bug with the `ws` package, but since they don't maintain their own type definitions, I'm opening the same bug report here.

**Comment by Methuselah96** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57590
